runtime/ffi.h

```diff
@@ -199,6 +252,14 @@ int64_t f$FFI$$sizeof(const CDataRef<T> &ref) {
   return sizeof(*ref.c_value);
 }
 
+template<class T1, class T2>
+int64_t f$FFI$$memcmp(T1 ptr1, T2 ptr2, int64_t size)  {
+  int64_t result = std::memcmp(ffi_c_value_ptr(ptr1), ffi_c_value_ptr(ptr2), size);
+  // contrary to what PHP documentation says, it actually only returns -1, 0 and 1 values,
+  // so we need to adjust memcmp results here
+  return std::clamp(result, -1L, 1L);
```

Comment:
Let's add `#include <algorithm>` at the file beginning to fix mac os build

Reply:
Huh. I wonder if it would be better to just do clamp operation inplace.
Is it idiomatic in our code base by the way to use clamp? I don't see it that often.

Reply:
+1 for in-place

Reply:
Done